rust/tool/src/generation.rs

```diff
@@ -40,10 +40,19 @@ pub struct Generation {
 impl Generation {
     pub fn from_link(link: &GenerationLink) -> Result<Self> {
         let bootspec_path = link.path.join("boot.json");
-        let generation: BootspecGeneration = serde_json::from_slice(
-            &fs::read(bootspec_path).context("Failed to read bootspec file")?,
-        )
-        .context("Failed to parse bootspec json")?;
+        let generation = fs::read(bootspec_path)
+            .with_context(|| "Failed to read bootspec")
+            .and_then(|v| serde_json::from_slice(&v).context("Failed to read bootspec JSON"))
+            // TODO: this should be much easier, add a From<GenerationVX> for BootspecGeneration
+            // this should enable us to do `into()` on the Result
+            // anyhow compatibility of bootspec would be nice too.
```

Comment:
I'll work on making real errors this weekend, hopefully will make this easier for you guys!